src/components/Select/index.tsx

```diff
@@ -0,0 +1,135 @@
+import { Form, Icons } from '@storybook/components'
+import { ThemeProvider } from '@storybook/theming'
+import { useCombobox } from 'downshift'
+import React, { memo, useState } from 'react'
+
+import { Item } from '../../types'
+import { Chip, Container, Menu, MenuItem, Remove, Root, Toggle } from './styled'
+import { isArray, isUndefined } from '../../utilities'
+
+export type Props = {
+    items: Item[]
+    valid?: 'valid' | 'error' | 'warn'
+} & (
+    | {
+          selected: Item | undefined
+          isMulti?: false
+          onChange: (item: Item | null) => void
+      }
+    | {
+          selected: Item[] | undefined
+          isMulti: true
+          onChange: (items: Item[]) => void
+      }
+)
+
+export const Select = memo((props: Props) => {
+    const { items, selected, isMulti, valid } = props
+    const selectedItems = isUndefined(selected)
+        ? []
+        : isArray(selected)
+        ? selected
+        : [selected]
+    const [filteredItems, setFilteredItems] = useState(items)
+    const {
+        getComboboxProps,
+        getInputProps,
+        getItemProps,
+        getLabelProps,
+        getMenuProps,
+        getToggleButtonProps,
+        highlightedIndex,
+        isOpen,
+        reset,
+    } = useCombobox({
+        items: filteredItems,
+        onInputValueChange: ({ inputValue }) => {
+            setFilteredItems(
+                items.filter(({ label }) =>
+                    label.toLowerCase().includes(inputValue.toLowerCase()),
+                ),
+            )
+        },
+        onSelectedItemChange: ({ selectedItem }) => {
+            if (selectedItem) {
+                update(
+                    isMulti
+                        ? Array.from(new Set([...selectedItems, selectedItem]))
+                        : [selectedItem],
+                )
+            }
+        },
+        onIsOpenChange: (state) => {
+            if (!state.isOpen) {
+                reset()
```

Comment:
Need to fix this

Reply:
Turns out that hovering/clicking on a label with a for targets the first interactive component